pkg/kubelet/cm/dra/plugin/registration.go

```diff
@@ -178,30 +186,35 @@ func (h *RegistrationHandler) RegisterPlugin(pluginName string, endpoint string,
 	return nil
 }
 
-func (h *RegistrationHandler) validateVersions(
-	pluginName string,
-	versions []string,
-) (*utilversion.Version, error) {
-	if len(versions) == 0 {
-		return nil, errors.New("empty list for supported versions")
+// validateSupportedServices identifies the highest supported gRPC service for
+// NodePrepareResources and NodeUnprepareResources and returns its name
+// (e.g. [drapbv1beta1.DRAPluginService]). An error is returned if the plugin
+// is unusable.
+func (h *RegistrationHandler) validateSupportedServices(pluginName string, supportedServices []string) (string, error) {
+	if len(supportedServices) == 0 {
+		return "", errors.New("empty list of supported gRPC services (aka supported versions)")
 	}
 
-	// Validate version
-	newPluginHighestVersion, err := utilversion.HighestSupportedVersion(versions)
-	if err != nil {
-		// HighestSupportedVersion includes the list of versions in its error
-		// if relevant, no need to repeat it here.
-		return nil, fmt.Errorf("none of the versions are supported: %w", err)
+	// Pick most recent version if available.
+	chosenService := ""
+	for _, service := range []string{
+		// Sorted by most recent first, oldest last.
+		drapbv1beta1.DRAPluginService,
+		drapbv1alpha4.NodeService,
+	} {
+		if slices.Contains(supportedServices, service) {
+			chosenService = service
+			break
+		}
 	}
 
-	existingPlugin := draPlugins.get(pluginName)
-	if existingPlugin == nil {
-		return newPluginHighestVersion, nil
+	// Fall back to alpha if necessary because
+	// plugins at that time didn't advertise gRPC services.
+	if chosenService == "" {
+		chosenService = drapbv1alpha4.NodeService
```

Comment:
At some point we will need to remove this logic. Maybe in separate PR we need to write a warning when this fallback was chosen.

Reply:
Agreed, this is temporary. Once all DRA drivers are known to support 1.32, we can remove it. It's support for an alpha, so we can do that from one release to the next, although it won't hurt to give it more time.